orientation: expose create_north_up_east_left/right

kecnry opened the pull request:
<!-- This comments are hidden when you submit the pull request,
so you do not need to remove them! -->

<!-- Please be sure to check out our code of conduct,
https://github.com/spacetelescope/jdaviz/blob/main/CODE_OF_CONDUCT.md . -->

### Description
<!-- Provide a general description of what your pull request does.
Complete the following sentence and add relevant details as you see fit. -->

<!-- In addition please ensure that the pull request title is descriptive
and allows maintainers to infer the applicable viz component(s). -->

This pull request exposes `create_north_up_east_left(...)` and `create_north_up_east_right(...)` methods via the public API of the Orientation plugin, along with API hints.

This also renames `set_on_create` to `set_as_orientation` in these methods, as it sets them whether or not they are created.  This does not however, rename the already public `set_on_create` within `add_orientation(...)`

Docs updates: [Orientation plugin API docs](https://jdaviz--3308.org.readthedocs.build/en/3308/api/jdaviz.configs.imviz.plugins.orientation.orientation.Orientation.html#jdaviz.configs.imviz.plugins.orientation.orientation.Orientation)

API hints:
<img width="701" alt="image" src="https://github.com/user-attachments/assets/6580ceff-ab67-40f8-81a5-670a5750e9d0">


<!-- If the pull request closes any open issues you can add this.
If you replace <Issue Number> with a number, GitHub will automatically link it.
If this pull request is unrelated to any issues, please remove
the following line. -->


### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)
<!-- This section is to be filled by package maintainer(s) who will
review this pull request. -->

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

kecnry commented on jdaviz/configs/imviz/plugins/orientation/orientation.py:
```diff
@@ -52,15 +52,20 @@ class Orientation(PluginTemplateMixin, ViewerSelectMixin):
     * ``east_left``
-    * ``add_orientation``
+    * :meth:`add_orientation`
+    * :meth:`create_north_up_east_left`
+    * :meth:`create_north_up_east_right`
```
should we remove `create_` from the method names since these can create AND/OR set?

bmorris3 replied:
I might vote for `set_` rather than `create_`, since `set_` tells you the end result?

kecnry replied:
unless you pass `set_as_orientation = False` 🤔... but I think I agree that I still prefer that

camipacifici commented:
Clear, works well for me.

bmorris3 commented on jdaviz/configs/imviz/plugins/orientation/orientation.py:
```diff
@@ -52,15 +52,20 @@ class Orientation(PluginTemplateMixin, ViewerSelectMixin):
     """
     template_file = __file__, "orientation.vue"
 
+    # defined as traitlet to allow access from UI - leave fixed
```
```suggestion
    # defined as traitlet to allow access from UI - leave fixed
    # since this name is used to identify WCS-only layers elsewhere in jdaviz.
```

kecnry replied:
I replaced the comment with: `defined as traitlet in addition to global variable above to allow access from UI - leave fixed` to give more context.  Maybe we should define this in one place and import into others if it is an assumed fixed string, but we can do that later 🤷‍♂️

bmorris3 reviewed (approved):
This looks great, thanks @kecnry!

kecnry commented:
@bmorris3 @camipacifici - I renamed `create_north_up_east_left/right` to `set_north_up_east_left/right` and also removed the `set_as_orientation` option from the public API... I don't see a reason why a user would want to create but not set the orientation and I think it only adds confusion, but if either of you want me to keep it, let me know and I'll revert that change.

camipacifici commented:
You are right, I am ok with that.
